### PR TITLE
R4R: Billy/1263 check bond bugs

### LINKED ISSUE #1263: Bonding Page Bugs
This looks like the same decimal problems we were having before...
screenshot 2018-09-03 14 24 30

(the one below is an easy fix and already taken care of)
screenshot 2018-09-03 13 58 25

### PR DESCRIPTION
**Pull request by okwme**
Closes #1263

Description: 🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞🐞

Atoms are shares are calculateTokens 😳

### CODE REVIEW

**Comment by okwme** [comment]
this fails coverage because of this open issue: https://github.com/cosmos/voyager/issues/1276

File: app/src/renderer/components/staking/PageBond.vue [hunk 1]
```diff
@@ -365,7 +371,7 @@ export default {
       return value + "%"
     },
     limitMax(delegate, max) {
-      delegate.atoms = parseFloat(delegate.atoms) // for stuff like 0-101-9
+      delegate.atoms = delegate.atoms // for stuff like 0-101-9
```
Comment (faboweb) [inline app/src/renderer/components/staking/PageBond.vue]: 1 = 1 ?

Comment (okwme) [reply app/src/renderer/components/staking/PageBond.vue]: oops 🤡

File: app/src/renderer/vuex/getters.js [hunk 2]
```diff
@@ -24,17 +26,29 @@ export const wallet = state => state.wallet
...
+  let totalOldBondedAtoms = new BN(0)
+  Object.keys(getters.delegation.committedDelegates).forEach(
+    delegatorAddress => {
+      let shares = getters.delegation.committedDelegates[delegatorAddress]
+      let delegator = getters.delegates.delegates.find(
+        d => d.id === delegatorAddress
+      )
+      if (!delegator) {
+        return
+        // throw new Error("no delegator with id: " + delegatorAddress)
```
Comment (faboweb) [inline app/src/renderer/vuex/getters.js]: why do we not throw here?

File: app/src/renderer/components/wallet/PageWallet.vue [hunk 3]
```diff
@@ -64,6 +65,11 @@ import ModalSearch from "common/TmModalSearch"
 import VmToolBar from "common/VmToolBar"
 export default {
   name: "page-wallet",
+  data: () => {
```
Comment (fedekunze) [inline app/src/renderer/components/wallet/PageWallet.vue]: data: () => num

File: app/src/renderer/scripts/common.js [hunk 4]
```diff
@@ -1,18 +1,56 @@
...
+//     .times(totalTokensD)
+//     .div(totalSharesD.times(totalTokensN))
+// }
+
+module.exports.calculateTokens = function(delegator, shares) {
+  // this is the based on the idea that tokens should equal
+  // (myShares / totalShares) * totalTokens where totalShares
+  // and totalTokens are both represented as fractions
+  let myShares = new BN(shares || 0)
+  // return myShares
```
Comment (fedekunze) [inline app/src/renderer/scripts/common.js]: delete comment

File: app/src/renderer/vuex/getters.js (hunk 2 above)
Comment (fedekunze) [inline app/src/renderer/vuex/getters.js]: delete comment

File: app/src/renderer/vuex/modules/delegation.js [hunk 5]
```diff
@@ -139,31 +143,36 @@ export default ({ node }) => {
...
+              delegation.delegate,
+              state.committedDelegates[delegation.delegate.owner]
+            ) - delegation.atoms
         )
         .reduce((sum, diff) => sum + diff, 0)
       commit("setAtoms", rootState.user.atoms + atomsDiff)
 
       // we optimistically update the committed delegations
-      updateCommittedDelegations(delegations, commit)
+      // updateCommittedDelegations(delegations, commit)
```
Comment (fedekunze) [inline app/src/renderer/vuex/modules/delegation.js]: delete comment

File: app/src/renderer/vuex/modules/delegation.js [hunk 6]
```diff
@@ -139,31 +143,36 @@ export default ({ node }) => {
...
       // TODO usually I would just query the new state through the LCD and update the state with the result, but at this point we still get the old shares
-      dispatch("updateDelegates").then(() =>
-        updateCommittedDelegations(delegations, commit)
-      )
+      setTimeout(async () => {
+        dispatch("updateDelegates") //.then(() =>
```
Comment (fedekunze) [inline app/src/renderer/vuex/modules/delegation.js]: delete comments

File: test/unit/specs/components/staking/PageBond.spec.js [hunk 7]
```diff
@@ -7,52 +7,58 @@ describe("PageBond", () => {
...
+    {
+      id: "pubkeyZ",
+      voting_power: 20000,
+      delegator_shares: "75000/1",
+      tokens: "20000/1",
+      shares: 75000
+    },
+    candidates[2] // this is the revoked one
+  )
+  chileanValidator.description.moniker = "aChileanValidator"
```
Comment (fedekunze) [inline test/unit/specs/components/staking/PageBond.spec.js]: 🇨🇱

File: test/unit/specs/components/staking/PageBond.spec.js [hunk 8]
```diff
@@ -80,19 +86,31 @@ describe("PageBond", () => {
 
   it("shows number of total atoms", () => {
     store.commit("setAtoms", 1337)
-    expect(wrapper.vm.totalAtoms).toBe(1437) // plus unbonding atoms
+    expect(wrapper.vm.totalAtoms).toBe("1437") // plus unbonding atoms
   })
 
   it("shows old bonded atoms ", () => {
+    // wrapper.vm.$store.getters.delegates = () => {
```
Comment (fedekunze) [inline test/unit/specs/components/staking/PageBond.spec.js]: delete comment

File: app/src/renderer/vuex/modules/delegation.js (hunk 6 above)
Comment (okwme) [reply app/src/renderer/vuex/modules/delegation.js]: https://github.com/cosmos/voyager/issues/1300leaving in to be picked up by

**Pull request merged**